### DISCUSSION TITLE
Trying to run kit.svelte.dev locally: `pnpm run update` fails

### DISCUSSION THREAD

**Discussion by nilslindemann** [sveltejs/kit#10298] [Q&A] [answered]
I want to locally test the effects of a PR to the [kit.svelte.dev](https://kit.svelte.dev/) site.

System info:

```
  System:
    OS: Linux 5.15 Linux Mint 21.1 (Vera)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 12.48 GB / 15.36 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.3.1 - ~/Node.js/v20.3.1/bin/node
    Yarn: 3.6.1 - ~/Node.js/v20.3.1/bin/yarn
    npm: 9.7.2 - ~/Node.js/v20.3.1/bin/npm
    pnpm: 8.6.5 - ~/Node.js/v20.3.1/bin/pnpm
```
([downloaded a recent Node](https://nodejs.org/en/download), added bin directory to .bashrc, `corepack enable`, `corepack prepare pnpm@latest --activate`, `corepack prepare yarn@stable --activate`)

The [kit repo](https://github.com/sveltejs/kit) is locally cloned to `~/Svelte/kit`. There I ran `pnpm install` as described [here](https://github.com/sveltejs/kit/blob/f7c0246a1dee8050b8a5defee86eb7c90242c692/CONTRIBUTING.md).

Now, when I run `pnpm run update` in `~/Svelte/kit/sites/kit.svelte.dev` like described [here](https://github.com/sveltejs/kit/blob/f7c0246a1dee8050b8a5defee86eb7c90242c692/sites/kit.svelte.dev/README.md), I get the following error message:

```
> kit.svelte.dev@0.0.1 update /home/nils/Svelte/kit/sites/kit.svelte.dev
> node scripts/check-doc-links.js && node scripts/types

unhandled JSDoc tag: deprecated
unhandled JSDoc tag: deprecated
unhandled JSDoc tag: deprecated
unhandled JSDoc tag: deprecated
node:fs:592
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/home/nils/Svelte/kit/packages/kit/types/index.d.ts'
    at Object.openSync (node:fs:592:3)
    at Object.readFileSync (node:fs:460:35)
    at read_d_ts_file (file:///home/nils/Svelte/kit/sites/kit.svelte.dev/scripts/types/index.js:203:17)
    at file:///home/nils/Svelte/kit/sites/kit.svelte.dev/scripts/types/index.js:239:15
    at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
    at async DefaultModuleLoader.import (node:internal/modules/esm/loader:246:24)
    at async loadESM (node:internal/process/esm_loader:40:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/nils/Svelte/kit/packages/kit/types/index.d.ts'
}

Node.js v20.3.1
 ELIFECYCLE  Command failed with exit code 1.
> 
```

I am not a node expert and could need some help with that error message. It seems something typescript related is missing. How would I add it?

**Comment by geoffrich** [comment, marked as answer]
Looks like something is missing from the setup instructions - you also need to run `pnpm check` in the root of the `kit` repo. This will generate the missing `types/index.d.ts` file that the site uses to generate documentation. I'll open a PR to update the contributing guide.

**Reply by geoffrich** [reply]
Opened https://github.com/sveltejs/kit/pull/10301

**Reply by nilslindemann** [reply]
Thank you very much, @geoffrich, this did the trick.
